src/pages/UnlockedWallet/AddressesPage/AddressCard.tsx

```diff
@@ -44,23 +46,23 @@ const AddressCard = ({ hash, className }: AddressCardProps) => {
   const navigate = useNavigate()
   const { t } = useTranslation()
   const theme = useTheme()
-  const { getAddress, mainAddress, updateAddressSettings } = useAddressesContext()
+  const [address, { name: walletName, mnemonic }] = useAppSelector((s) => [
+    selectAddressByHash(s, hash),
+    s.activeWallet
+  ])
   const { data: price, isLoading: isPriceLoading } = useGetPriceQuery(currencies.USD.ticker)
 
-  const address = getAddress(hash)
+  if (!address || !walletName || !mnemonic) return null
 
-  if (!address) return null
-
-  const alphBalance = parseFloat(convertSetToAlph(BigInt(address.details.balance)))
+  const alphBalance = parseFloat(convertSetToAlph(BigInt(address.balance)))
   const fiatBalance = alphBalance * (price ?? 0)
   // TODO: Fetch tokens from explorer API and store in Redux
-  const tokens: string[] = address.details.balance !== '0' ? ['ALPH'] : []
+  const tokens: string[] = address.balance !== '0' ? ['ALPH'] : []
 
   const navigateToAddressDetailsPage = (hash: AddressHash) => () => navigate(`/wallet/addresses/${hash}`)
 
   const setAsDefaultAddress = (event: MouseEvent<HTMLDivElement>) => {
-    mainAddress && updateAddressSettings(mainAddress, { ...mainAddress?.settings, isMain: false })
-    updateAddressSettings(address, { ...address.settings, isMain: true })
+    changeDefaultAddress(address, { walletName, mnemonic })
```

Comment:
Once we figure out this "data key", our code of updating encrypted settings will become simpler.

Reply:
💯